src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -3486,7 +3486,24 @@ class TableExpressionSegment(ansi.TableExpressionSegment):
         # Nested Selects
         Bracketed(Ref("SelectableGrammar")),
         Ref("ValuesClauseSegment"),
-        Ref("StagePath"),
+        Sequence(
+            Ref("StagePath"),
+            Bracketed(
+                Delimited(
+                    Sequence(
+                        "FILE_FORMAT",
+                        Ref("ParameterAssignerSegment"),
+                        Ref("FileFormatSegment"),
+                    ),
+                    Sequence(
+                        "PATTERN",
```

Comment:
By structuring it this way, are we forcing `pattern` to come after `file_format`?

Reply:
No we are not. Added a test case to show this.

Not sure if it's technically valid to be in either order (which is why I didn't add a test case explicitly until now), but we support it.